### ISSUE TITLE
[ FEATURE ] English language fallback for episode names 

### ISSUE THREAD

**Issue by FireBladeNooT** [pymedusa/Medusa#1891]
**What you did: Update french database
**What happened: Some episodes have empty name when french name doesn'exist in tvdb
**What you expected: Episodes should get english name

**Logs:**
```
2017-01-03 12:33:19 INFO     SHOWQUEUE-UPDATE :: [677b47a] 300835: Ajin S01E01 has no name on TVDBv2. Setting to an empty string
```

I try with Default Indexer Language in english and in french

**Comment by labrys** [comment]
I agree, the user should be able to set a fall-back language (or list thereof) that could be used to populate missing data.  This would be greatly simplified if TVDB provided a list of languages that have data for a record, but unfortunately they don't.

**Comment by OmgImAlexis** [comment]
@labrys is there not a way to get the original produced language for the show from tvdb? If so we could use that as a fall back.

**Comment by labrys** [comment]
The closest option is to take the network, figure out what country it's from, then use that Country to provide a language-hint.  However the network information provided by TVDB is just the name and does contain any information, so we would likely need to get that from our network timezones file.

TLDR: No not really.

**Comment by OmgImAlexis** [comment]
We could always add yet another JSON file for mapping shows/networks to languages?

**Comment by labrys** [comment]
IMO best way to handle it would be fall back to other indexers for data not provided by TVDB.  For example given a show with a TVMaze mapping: the TVMaze network includes the country and timezone, so instead of using our network timezones json use the timezone from TVMaze's network, for any TVDB show that doesn't provide the users prefered language, use the country from the TVMaze network to try that language as the fallback language on TVDB, etc.

**Comment by OmgImAlexis** [comment]
That may actually work since the main difference between indexers is episode mapping, most other details seems to be interchangeable.

**Comment by labrys** [comment]
Well episodes don't always map on a 1-1 basis and series don't always match on a 1-1 basis (for example a series on TVDB may be two series on TVMaze (e.g. Spartan)).  If they did then you could just pull episode titles from another indexer.  Those types of mismatches is why you'd have to do the "get shows country from other indexers network to figure out this indexers primary metadata language for the show" type of acrobatics.

**Comment by p0psicles** [comment]
The issue here, is something different. As he mentioned that he tried changing the language to english and it still didn't show the episode names. While I can confirm that it should do this.

**Comment by FireBladeNooT** [comment]
@p0psicles 
I change the default language but show langage still stay in french, that's why I've nothing.
I've miss understood the default language, it's only used when we add a show, it change the language propose by default.

**Comment by p0psicles** [comment]
I asume that your editing the show, switch the french language to english and click on submit, wait like 5 minutes max, and then refresh the page?

After that it still has the french language right?

Can you perform these actions, and then share the full debug log with me?

**Comment by FireBladeNooT** [comment]
Ok, I'll do that.

**Comment by FireBladeNooT** [comment]
So, I test the switch of langage and english names appears if I switch langage.
But it disappears if I come back in french.

In the log:
````
2017-01-12 20:14:34 INFO     SHOWQUEUE-UPDATE :: [677b47a] 176941: Sherlock S00E08 has no name on TVDBv2. Setting to an empty string
````

**Comment by labrys** [comment]
Closing this as it has been added to the  #282 **Master Feature List** to track all feature requests.  Discussion specific to this feature can continue in this thread.

**Comment by fegyosz** [comment]
This is still a missing feature but not just for episode names. The same should be applied to other metadata files like posters. Posters are many times uploaded in English only while the series has a localized name. I need to choose between localized names or having a poster.
